spec/helpers/application_helper/buttons/new_host_aggregate_spec.rb

```diff
@@ -1,19 +1,38 @@
 describe ApplicationHelper::Button::NewHostAggregate do
+  include Spec::Support::RBACHelper
+
   let(:view_context) { setup_view_context_with_sandbox({}) }
-  let(:button) { described_class.new(view_context, {}, {}, {}) }
+  let(:button)       { described_class.new(view_context, {}, {}, {}) }
+
+  before do
+    setup_user_with_tenant
+  end
+
+  let!(:provider) { FactoryGirl.create(:ems_openstack, :tenant => @tenant) }
```

Comment:
Please don't use the `!` here, rather force the `provider` creation in a before block when necessary.
Also the `@tenant` seems scary, you should not have instance variables as you can use `let`...